### PR TITLE
feat(timeline artifact): add invocation trace events.

### PR DESCRIPTION
**Pull request by asafkorem**
Add invocation (actions and expectations) started & ended events, along with their stack trace.

As addressed in #3536.

### CODE REVIEW

**Comment by asafkorem** [comment]
@jonathanmos I made some changes, most of them as agreed f2f, please review 🙏🏼 🙂

File: detox/src/android/core/NativeElement.js [hunk 1]
```diff
@@ -32,56 +32,81 @@ class NativeElement {
...
-    return await new ActionInteraction(this._invocationManager, this, new actions.LongPressAction()).execute();
+    const action = new actions.LongPressAction();
+    const traceDescription = actionDescription.longPress();
+    return await new ActionInteraction(this._invocationManager, this, action, traceDescription).execute();
   }
 
   async multiTap(times) {
-    return await new ActionInteraction(this._invocationManager, this, new actions.MultiClickAction(times)).execute();
+    const action = new actions.MultiClickAction(times);
+    const traceDescription = actionDescription.multiTap();
```
Comment (jonathanmos) [inline detox/src/android/core/NativeElement.js:54]: multiTap is missing times arg

File: detox/src/android/interactions/native.js [hunk 2]
```diff
@@ -1,53 +1,63 @@
...
+      const resultObj = await this._invocationManager.execute(this._call);
+      return resultObj ? resultObj.result : undefined;
+    });
   }
 }
 
 class ActionInteraction extends Interaction {
-  constructor(invocationManager, element, action) {
-    super(invocationManager);
+  constructor(invocationManager, element, action, description) {
```
Comment (jonathanmos) [inline detox/src/android/interactions/native.js:30]: we need to change to traceDescription to match the rest

File: detox/src/ios/expectTwo.js [hunk 3]
```diff
@@ -287,22 +339,21 @@ class Element {
     return invocation;
   }
 
-  withAction(action, ...params) {
+  async withAction(action, description, ...params) {
     const invocation = this.createInvocation(action, null, ...params);
```
Comment (jonathanmos) [inline detox/src/ios/expectTwo.js:343]: need to change these to traceDescription to match the rest

File: detox/src/ios/expectTwo.js [hunk 4]
```diff
@@ -287,22 +339,21 @@ class Element {
...
-    return this._invocationManager.execute(invocation);
+    return await _executeInvocation(this._invocationManager, invocation, description);
   }
 
-  withActionAndTargetElement(action, targetElement, ...params) {
+  async withActionAndTargetElement(action, targetElement, description, ...params) {
     const invocation = this.createInvocation(action, targetElement, ...params);
```
Comment (jonathanmos) [inline detox/src/ios/expectTwo.js:348]: same as previous: description -> traceDescription

**Review by jonathanmos — CHANGES_REQUESTED** [review]

File: detox/src/android/espressoapi/EspressoDetox.js [hunk 5]
```diff
@@ -52,7 +52,7 @@ class EspressoDetox {
   }
 
   static setURLBlacklist(urls) {
-    if (typeof urls !== 'object' || !Array.isArray(urls)) {
+    if (typeof urls !== 'object' || !urls instanceof Array) {
```
Comment (asafkorem) [inline detox/src/android/espressoapi/EspressoDetox.js:55]: Why?

Comment (d4vidi) [reply detox/src/android/espressoapi/EspressoDetox.js:55]: what about `_.isArray`, `_.isObject`?

Comment (asafkorem) [reply detox/src/android/espressoapi/EspressoDetox.js:55]: That's autogenerated code actually 😅

**Comment by asafkorem** [comment]
Squashed commits with meaningful commit messages.

**Review by jonathanmos — APPROVED** [review]

**Review by jonathanmos — APPROVED** [review]

**Pull request merged**